skimage/transform/_geometric.py

```diff
@@ -1446,8 +1448,13 @@ def estimate(self, src, dst, order=2):
 
         A[:rows, -1] = xd
         A[rows:, -1] = yd
-
-        _, _, V = np.linalg.svd(A)
+        
+        if A.shape[1] <= A.shape[0]:
+            # scipy sparse svd uses less memory
+             _, _, V = svds(A, k=1, which='SM', return_singular_vectors='vh')
```

Comment:
In the sparse case, you might want to consider setting up A as a sparse matrix to further reduce memory usage.